terraform/modules/vpc/public.tf

```diff
@@ -12,12 +12,19 @@ resource "aws_subnet" "public" {
   tags                    = merge(var.tags, { Name = "${var.name}-public-${local.zone_names[count.index]}" })
 }
 
-resource "aws_default_route_table" "public" {
-  default_route_table_id = module.vpc.vpc.main_route_table_id
-  tags                   = merge(var.tags, { Name = "${var.name}-public" })
+resource "aws_route_table" "public" {
+  vpc_id = module.vpc.vpc.id
+  tags   = merge(var.tags, { Name = "${var.name}-public" })
+}
+
+resource "aws_route_table_association" "public" {
+  count          = local.zone_count
+  route_table_id = aws_route_table.public.id
+  subnet_id      = aws_subnet.public[count.index].id
+}
 
-  route {
-    cidr_block = "0.0.0.0/0"
-    gateway_id = aws_internet_gateway.igw.id
-  }
+resource "aws_route" "public" {
+  route_table_id         = aws_route_table.public.id
+  destination_cidr_block = "0.0.0.0/0"
```

Comment:
didn't our Security Architect want this range reducing? Is that something for another piece of work?

Reply:
No can do boss.  A subnet is no longer public if it doesn't have a "0.0.0.0/0" route out to either a NGW or IGW. So...we'll route everything to the Internet but the SGs should mean its blocked anyway.